torch/csrc/utils/tensor_flatten.h

```diff
@@ -4,9 +4,16 @@
 #include <torch/csrc/WindowsTorchApiMacro.h>
 #include <ATen/ATen.h>
 #include <utility>
+#include <c10/core/TensorOptions.h>
 
 namespace torch { namespace utils {
 
+inline int64_t type_id(const at::Tensor& tensor) {
+  return static_cast<int64_t>(tensor.options().backend()) *
+      static_cast<int64_t>(at::ScalarType::NumOptions) +
+      static_cast<int64_t>(tensor.scalar_type());
+}
```

Comment:
Could you explain how this works to generate a unique type id? Adding a comment here would also be helpful.

Reply:
it's probably not unique for all types, just for what is necessary to re-order tensors for flattening. Added a comment to explain